nifi-nar-bundles/nifi-extension-utils/nifi-event-transport/src/main/java/org/apache/nifi/event/transport/netty/NettyEventSender.java

```diff
@@ -25,20 +25,31 @@
 import org.apache.nifi.event.transport.EventSender;
 
 import java.net.SocketAddress;
+import java.util.concurrent.TimeUnit;
 
 /**
  * Netty Event Sender with Channel Pool
  *
  * @param <T> Event Type
  */
 class NettyEventSender<T> implements EventSender<T> {
+    static final long SHUTDOWN_QUIET_PERIOD = 2000;  // netty default
+    static final long SHUTDOWN_TIMEOUT = 15000;  // netty default
+    static final TimeUnit SHUTDOWN_TIME_UNIT = TimeUnit.MILLISECONDS;
+
     private final EventLoopGroup group;
 
     private final ChannelPool channelPool;
 
     private final SocketAddress remoteAddress;
 
-    private boolean singleEventPerConnection;
+    private final boolean singleEventPerConnection;
+
+    private final long shutdownQuietPeriod;
+
+    private final long shutdownTimeout;
+
+    private final TimeUnit shutdownTimeUnit;
```

Comment:
Rather than having separate properties, it seems like the implementation could be streamlined using `java.time.Duration`.  This would remove the need for the `shutdownTimeUnit` property.

Reply:
I was originally modeling after the netty API.  But this suggestion does reduce the complexity; thanks.